desktop/src/navigator.rs

```diff
@@ -106,8 +106,20 @@ impl NavigatorBackend for ExternalNavigatorBackend {
     }
 
     fn fetch(&self, request: Request) -> OwnedFuture<Response, Error> {
+        // Check for relative URL and act as Flash would and use "/", otherwise pass result //
+        let req_parsed_url = match Url::parse(request.url()) {
+            Ok(url) => Ok(url),
+            Err(e) => {
+                if e == ParseError::RelativeUrlWithoutBase && !request.url().starts_with('/') {
+                    self.movie_url.join(format!("/{}", request.url()).as_str())
+                } else {
+                    self.movie_url.join(request.url())
+                }
+            }
+        };
+
         // TODO: honor sandbox type (local-with-filesystem, local-with-network, remote, ...)
-        let full_url = match self.movie_url.join(request.url()) {
```

Comment:
Thanks for your first contribution!

I think the root cause of #8315 is that the desktop backend stores the entire SWF URL, whereas the web backend stores just its containing directory. So rather I'd like to align to the web backend instead. The easiest way I found to convert a full `Url` to its containing directory is using [`url::PathSegmentsMut::pop`](https://docs.rs/url/latest/url/struct.PathSegmentsMut.html#method.pop):
```rs
let mut url = Url::parse("https://example.com/path/to/game.swf")?;
url.path_segments_mut().map_err(|_| "cannot be base")?.pop();
assert_eq!(url.as_str(), "https://example.com/path/to/");
```

Reply:
@relrelb 
The ? cannot be used on a non-result returning function, the constructor returns a `Self` (`ExternalNavigatorBackend`) and not a Result. In Desktop, the URL gets checked before the `ExternalNavigatorBackend::new` for an error, and shuts down if it there is one. The `::new` could be modified to accept only a base URL that is preprocessed that way? Or could just 'panic' (use unwrap) for it, 'cause the URL is checked anyway.

Reply:
To be compatible with Web, we could just do the 'panic' (`unwrap()`) route, which I've already tested and works.

Reply:
(And I'd go with `url.path_segments_mut().unwrap().clear();` if that's alright?)